apps/blur/halide_blur_generator.cpp

```diff
@@ -15,8 +40,50 @@ class HalideBlur : public Halide::Generator<HalideBlur> {
         blur_y(x, y) = (blur_x(x, y) + blur_x(x, y+1) + blur_x(x, y+2))/3;
 
         // How to schedule it
-        blur_y.split(y, y, yi, 8).parallel(y).vectorize(x, 8);
-        blur_x.store_at(blur_y, y).compute_at(blur_y, yi).vectorize(x, 8);
+        if (get_target().has_gpu_feature()) {
+            // GPU schedule.
+            switch (schedule) {
+            case BlurGPUSchedule::Inline:
+                // - Fully inlining.
+                blur_y.gpu_tile(x, y, tile_x, tile_y);
+                break;
+            case BlurGPUSchedule::Cache:
+                // - Cache blur_x calculation.
+                blur_y.gpu_tile(x, y, tile_x, tile_y);
+                blur_x.compute_at(blur_y, Var::gpu_blocks()).gpu_threads(x, y);
+                break;
+            case BlurGPUSchedule::Slide: {
+                // - Instead caching blur_x calculation explicitly, the
+                //   alternative is to allow each work-item in OpenCL or thread
+                //   in CUDA to calculate more rows of blur_y so that temporary
+                //   blur_x calculation is re-used implicitly. This achieves
+                //   the similar schedule of sliding window.
+                Var yi("yi");
+                blur_y.split(y, y, yi, tile_y);
```

Comment:
Our usual convention for chaining scheduling calls across multiple lines is to not repeat the Func name:

    blur_y.split(...)
        .reorder(...)
        .unroll(...)
        .gpu_tile(...);